### ISSUE TITLE
ModernBERT: padded batch items return all-NaN embeddings when the longest sequence is a multiple of 32 (>= 128 tokens)

### ISSUE THREAD

**Issue by builtbycnob** [Blaizzy/mlx-embeddings#80]
Any mixed-length batch through a ModernBERT model returns all-NaN vectors for every padded item whenever the longest item in the batch tokenizes to an exact multiple of 32, at or above 128. The longest item itself is fine. Equal-length batches are fine. Single texts are always fine. The response is otherwise normal, so downstream code gets silently corrupt embeddings.

Found this in production behind an inference server. It reproduces with mlx-embeddings alone.

**Repro** (mlx-embeddings 0.1.0, mlx 0.31.2 and 0.32.0, macOS 26, M3 Ultra; model `nomic-ai/modernbert-embed-base`, fp16):

```python
import math
from mlx_embeddings.utils import load, generate

model, processor = load("nomic-ai/modernbert-embed-base")
tok = processor._tokenizer if hasattr(processor, "_tokenizer") else processor

def with_tokens(n):
    t = "documento " * ((n - 3) // 2)
    while len(tok.encode(t)) < n:
        t += "e "
    return t

short = "short probe sentence"
for n in (255, 256, 257, 288):
    out = generate(model, processor, texts=[short, with_tokens(n)])
    bad = [sum(1 for x in v.tolist() if not math.isfinite(x)) for v in out.text_embeds]
    print(f"longest={n} tokens -> non-finite per vector: {bad}")
```

Output:

```
longest=255 tokens -> non-finite per vector: [0, 0]
longest=256 tokens -> non-finite per vector: [768, 0]
longest=257 tokens -> non-finite per vector: [0, 0]
longest=288 tokens -> non-finite per vector: [768, 0]
```

Measured boundary: 128, 224, 256, 288, 320, 512 and 640 are all toxic; 32, 64, 255, 257, 259 and 289 are clean. In a batch of 3 with one long item, both padded items go NaN.

**Mechanism**, as far as I traced it in `mlx_embeddings/models/modernbert.py`:

`_update_attention_mask` builds the additive masks with `-1e9` and then casts them back to `attention_mask.dtype`, which is the tokenizer's int32. `mx.fast.scaled_dot_product_attention` promotes that int32 mask to the model's float16, where -1e9 overflows to -inf. ModernBERT alternates global attention with sliding-window local attention (window 128). A padded query row whose whole window lies inside padding gets an all -inf score row, softmax returns NaN, and the next layer's 0 * NaN spreads it across the padded sequence. The multiple-of-32 length condition selects the SDPA kernel tile path that exposes it; off-multiple lengths mask the same rows without producing NaN.

This looks like the same family as #72 / #73: an additive mask cast to the wrong dtype breaking padded items, per architecture.

**Suggested fix**: build the additive masks in the model compute dtype with a finite fp16-safe constant (for example `-6e4`, or `finfo.min / 2`) instead of `-1e9`, and never cast masks to integer dtypes. Guarding fully-masked query rows would close the NaN at the source regardless of dtype.
